packages/plugins/src/plugins/aave/lendingPoolsJobV3.ts

```diff
@@ -0,0 +1,132 @@
+import 'reflect-metadata';
+import {
+  borrowLendRatesPrefix,
+  formatTokenAddress,
+} from '@sonarwatch/portfolio-core';
+import { formatReservesAndIncentives } from '@aave/math-utils-v3';
+import { Cache } from '../../Cache';
+import { Job, JobExecutor } from '../../Job';
+import {
+  aave3platformId,
+  lendingPoolsPrefix,
+  v3lendingConfigs,
+} from './constants';
+import { LendingDataV3 } from './types';
+import { getRates, getTokenPriceSourceFromReserve } from './helpers';
+import { getUiProvidersV3 } from './helpersV3';
+
+const PLATFORM_ID = aave3platformId;
+
+/*
+ * Essentially the same as lendingPoolsJob.ts, but uses an updated library for v3.
+ * The previous library throws an error with v3 contracts.
+ */
+const executor: JobExecutor = async (cache: Cache) => {
+  const lendingConfigsArray = Array.from(v3lendingConfigs.values()).flat();
+
+  for (let i = 0; i < lendingConfigsArray.length; i++) {
+    const lendingConfig = lendingConfigsArray[i];
+    if (!lendingConfig) continue;
+
+    const { lendingPoolAddressProvider, chainId, networkId, elementName } =
+      lendingConfig;
+
+    const { poolDataProvider, incentiveDataProvider } = getUiProvidersV3(
+      networkId,
+      lendingConfig
+    );
+
+    if (!poolDataProvider || !incentiveDataProvider) continue;
+
+    // If it fails, this is probably because of a contract upgrade.
+    // Make sure contracts in /constants.ts are up-to-date.
```

Comment:
```suggestion
    // Make sure contracts in ./constants.ts are up-to-date.
```
We're referring to the file at the same level, correct ?